sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubAuthenticationPolicy.cs

```diff
@@ -28,9 +29,15 @@ public override void OnSendingRequest(HttpMessage message)
             var now = DateTimeOffset.UtcNow;
             var expiresAt = now + TimeSpan.FromMinutes(5);
 
-            var keyBytes = Encoding.UTF8.GetBytes(_credential.Key);
+            var key = _credential.Key;
+            var cache = _keyCache;
+            if (!ReferenceEquals(key, cache.Key))
```

Comment:
Do you need to handle cache == null here?

Reply:
I am pretty sure I do :-)